doc/storagemond/storagemond-hld.md

```diff
@@ -0,0 +1,529 @@
+
+`storage.devices` would contain:
+```
+{
+    'mmcblk0': <Emmcutil object>,
+    'sda': <SsdUtil object>
+}
+```
+
+we would then get static and dynamic information by leveraging the respective member function implementations of `SsdUtil` and `EmmcUtil`, as they both derive from `SsdBase`.
+We then leverage the following proposed `STATE_DB` schema to store and stream information about each of these disks.
+
+
+**NOTE:** <br>
+**Full support** -- monitors all the attributes mentioned in [section 2](#2-data-collection)<br>
+**Limited support** -- Support unavailable for Dynamic fields mentioned in [section 2.1](#21-priority-0-attributes)<br>
+**Not currently supported** -- Class currently unimplemented, no object created. No monitoring currently available.<br>
+
+<sub>UsbUtil and NVMeUtil classes are not yet available. EmmcUtil class does not currently support disk IO reads, disk IO writes and Reserved Blocks.</sub>
+
+#### **2.4.3 Support for common implementations**
+
+Specific data, such as Filesystem Input/Output (FS IO) Reads/Writes, can be uniformly collected regardless of the storage disk type, as it is extracted from files generated by the Linux Kernel. To streamline the process of gathering this information, we propose the implementation of a new parent class `StorageCommon()`, from which classes such as SsdUtil, EmmcUtil, USBUtil, and NVMUtil would inherit in addition to `SsdBase` (to be renamed `StorageBase`). This proposed class will reside in the `src/sonic-platform-common/sonic_platform_base/sonic_ssd` directory, in `storage_common.py`. The `StorageCommon()` class will have the following functions:
+
+```
+def get_fs_io_reads(self):
+    """
+    Function to get the latest reads on each disk by parsing the /proc/diskstats file
+
+    Returns:
+        The total number of FSIO reads
+    
+    Args:
+        N/A
+    """
+
+def get_fs_io_writes(self):
+    """
+    Function to get the latest writes on each disk by parsing the /proc/diskstats file
+
+    Returns:
+        The total number of FSIO writes
+    
+    Args:
+        N/A
+    """
+```
+
+**Logic for StorageCommon() get_fs_io_reads and get_fs_io_writes functions:**
+
+These two functions, `get_fs_io_reads` and `get_fs_io_writes`, are designed to retrieve the latest disk reads and writes, respectively, by parsing the `/proc/diskstats` file. They utilize similar logic, differing only in the column index used to extract the relevant information.
+
+1. **Check for `psutil` Module**:
+   - The functions first check if the `psutil` module is available in the current environment by examining the `sys.modules` dictionary.
+
+2. **Use `psutil` Module (if available)**:
+   - If `psutil` is available:
+     - The functions retrieve disk I/O counters, specifying the disk for which to get the counters.
+     - They then get the read or write count for the specified disk using `read_count` or `write_count` respectively.
+
+3. **Fallback to Parsing Disk Stats File**:
```

Comment:
@assrinivasan Master branch has psutil, so i think this fallback is not requried

Reply:
I have added a PR for pmon container to have psutil. That being said, I think this is a good fail-safe to have.

Reply:
Removed reference to fallback.